x-pack/plugins/observability_solution/observability_ai_assistant_management/public/routes/components/settings_tab/product_doc_entry.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
+import {
+  EuiButton,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../../hooks/use_kibana';
+import { useGetProductDocStatus } from '../../../hooks/use_get_product_doc_status';
+import { useInstallProductDoc } from '../../../hooks/use_install_product_doc';
+import { useUninstallProductDoc } from '../../../hooks/use_uninstall_product_doc';
+
+export function ProductDocEntry() {
+  const { overlays } = useKibana().services;
+
+  const [isInstalled, setInstalled] = useState<boolean>(true);
+  const [isInstalling, setInstalling] = useState<boolean>(false);
+
+  const { mutateAsync: installProductDoc } = useInstallProductDoc();
+  const { mutateAsync: uninstallProductDoc } = useUninstallProductDoc();
+  const { status, isLoading: isStatusLoading } = useGetProductDocStatus();
```

Comment:
Yeah, the installation is "stateless" on the server (we can't keep track of the installation process basically), as it was way easier in terms of implementation (no need to update some state, or check it, or handle concurrency...)

But yeah apparently switching tab resets the UI state, means the button is then available again if the tab is reloaded before the installation ends.

Doing that improvement makes sense, but it's not that trivial. so we might want let it go for the initial PR and to do it as a post FF bug fix / follow-up

Reply:
This is another area where it would be useful to utilize task manager, but you'll be blocked by this issue: https://github.com/elastic/kibana/issues/190661